server/services/v1/workers/worker.go

```diff
@@ -287,6 +296,68 @@ func (w *Worker) buildIndexTask(queueItem *metadata.QueueItem) error {
 	return tx.Commit(ctx)
 }
 
+func (w *Worker) buildSearchTask(queueItem *metadata.QueueItem) error {
+	var task metadata.IndexBuildTask
+	if err := jsoniter.Unmarshal(queueItem.Data, &task); err != nil {
+		return err
+	}
+
+	ctx := context.Background()
+	dbBranch := metadata.NewDatabaseNameWithBranch(task.ProjName, task.Branch)
+	tenant, err := w.tenantMgr.GetTenant(ctx, task.NamespaceId)
+	if err != nil {
+		return err
+	}
+
+	project, err := tenant.GetProject(task.ProjName)
+	if err != nil {
+		return err
+	}
+
+	db, err := project.GetDatabase(dbBranch)
+	if err != nil {
+		return err
+	}
+	coll := db.GetCollection(task.CollName)
+
+	req := &api.BuildCollectionSearchIndexRequest{
+		Branch:     task.Branch,
+		Collection: task.CollName,
+		Project:    task.ProjName,
+	}
+
+	qr := database.NewQueryRunnerFactory(w.txMgr, nil, w.searchStore)
```

Comment:
We can have this inside worker.

Reply:
I don't quite follow what you mean?

Reply:
I mean we can create it when we are creating the worker.